extmod/modbluetooth_nimble.c

```diff
@@ -614,16 +614,24 @@ STATIC int gap_scan_cb(struct ble_gap_event *event, void *arg) {
     return 0;
 }
 
-int mp_bluetooth_gap_scan_start(int32_t duration_ms) {
+int mp_bluetooth_gap_scan_start(int32_t duration_ms, int32_t interval_ms, int32_t window_ms) {
     if (duration_ms == 0) {
         duration_ms = BLE_HS_FOREVER;
     }
-    STATIC const struct ble_gap_disc_params discover_params = {
-        .itvl = BLE_GAP_SCAN_SLOW_INTERVAL1,
-        .window = BLE_GAP_SCAN_SLOW_WINDOW1,
+    uint16_t interval = BLE_GAP_SCAN_SLOW_INTERVAL1;
+    uint16_t window = BLE_GAP_SCAN_SLOW_WINDOW1;
+    if (interval_ms > 0) {
+        interval = MAX(BLE_HCI_SCAN_ITVL_MIN, MIN(BLE_HCI_SCAN_ITVL_MAX, interval_ms * 1000 / BLE_HCI_SCAN_ITVL));
+    }
+    if (window_ms > 0) {
+        window = MAX(BLE_HCI_SCAN_WINDOW_MIN, MIN(BLE_HCI_SCAN_WINDOW_MAX, window_ms * 1000 / BLE_HCI_SCAN_ITVL));
+    }
+    struct ble_gap_disc_params discover_params = {
+        .itvl = interval,
+        .window = window,
         .filter_policy = BLE_HCI_CONN_FILT_NO_WL,
         .limited = 0,
-        .passive = 0,
+        .passive = 1,  // TODO: Handle BLE_HCI_ADV_RPT_EVTYPE_SCAN_RSP in gap_scan_cb above.
```

Comment:
Why does this need to be changed?

Reply:
It was wrong before, because we're not handling the scan response event. Now that this change allows us to scan much faster, no point spamming the radio with active scans if we're going to ignore them.

I plan to add an `active=True` arg to scan() later when implementing active scans.